kratos/tests/geometries/test_triangle_3d_3.cpp

```diff
@@ -168,5 +168,64 @@ namespace Testing {
     KRATOS_CHECK(geom->IsInside(PointInEdge, LocalCoords, EPSILON));
   }
 
+
+    /** Tests the Jacobian determinants using 'GI_GAUSS_1' integration method.
+     * Tests the Jacobian determinants using 'GI_GAUSS_1' integration method.
+     */
+    KRATOS_TEST_CASE_IN_SUITE(Triangle3D3DeterminantOfJacobianArray1, KratosCoreGeometriesFastSuite) {
+      auto geom = GenerateRightTriangle3D3<Node<3>>();
+      const double ExpectedJacobian = 1.0;
+
+      Vector JacobianDeterminants;
+      geom->DeterminantOfJacobian( JacobianDeterminants, GeometryData::GI_GAUSS_1 );
+
+      for (unsigned int i=0; i<JacobianDeterminants.size(); ++i)
+      {
+          KRATOS_CHECK_NEAR(JacobianDeterminants[i], ExpectedJacobian, TOLERANCE);
+      }
+    }
+
+    /** Tests the Jacobian determinants using 'GI_GAUSS_2' integration method.
+     * Tests the Jacobian determinants using 'GI_GAUSS_2' integration method.
+     */
+    KRATOS_TEST_CASE_IN_SUITE(Triangle3D3DeterminantOfJacobianArray2, KratosCoreGeometriesFastSuite) {
+      auto geom = GenerateRightTriangle3D3<Node<3>>();
+      const double ExpectedJacobian = 1.0;
+
+      Vector JacobianDeterminants;
+      geom->DeterminantOfJacobian( JacobianDeterminants, GeometryData::GI_GAUSS_2 );
+
+      for (unsigned int i=0; i<JacobianDeterminants.size(); ++i)
+      {
+          KRATOS_CHECK_NEAR(JacobianDeterminants[i], ExpectedJacobian, TOLERANCE);
+      }
+    }
+
+    /** Tests the Jacobian determinants using 'GI_GAUSS_1' integration method.
+     * Tests the Jacobian determinants using 'GI_GAUSS_1' integration method.
+     */
+    KRATOS_TEST_CASE_IN_SUITE(Triangle3D3DeterminantOfJacobianIndex1, KratosCoreGeometriesFastSuite) {
+      auto geom = GenerateRightTriangle3D3<Node<3>>();
+      const double ExpectedJacobian = 1.0;
+
+      double JacobianDeterminant = geom->DeterminantOfJacobian( 1, GeometryData::GI_GAUSS_1 );
+      KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+    }
+
+    /** Tests the Jacobian determinants using 'GI_GAUSS_2' integration method.
+     * Tests the Jacobian determinants using 'GI_GAUSS_2' integration method.
+     */
+    KRATOS_TEST_CASE_IN_SUITE(Triangle3D3DeterminantOfJacobianIndex2, KratosCoreGeometriesFastSuite) {
+      auto geom = GenerateRightTriangle3D3<Node<3>>();
+      double JacobianDeterminant = 0.0;
+      const double ExpectedJacobian = 1.0;
+
+      JacobianDeterminant = geom->DeterminantOfJacobian( 1, GeometryData::GI_GAUSS_2 );
+      KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+
+      JacobianDeterminant = geom->DeterminantOfJacobian( 2, GeometryData::GI_GAUSS_2 );
+      KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
```

Comment:
do not make TOLERANCE to be global to the file

also, it is not a kratos var, so i would not make it all capital (this is of minor importance)